net/calendar-contacts-server/Portfile

```diff
@@ -0,0 +1,535 @@
+                ${f_or_d} \
+                ${f_or_d}.previous
+        }
+        if { [variant_isset "initialize"]
+             || ![file exists ${f_or_d}]
+            } {
+            if { [file isfile ${f_or_d}.macports] } {
+                xinstall -m 0644 \
+                    ${f_or_d}.macports \
+                    ${f_or_d}
+            } elseif { [file isdirectory ${f_or_d}.macports] } {
+                xinstall -m 0755 -d ${f_or_d}
+                foreach f [glob -nocomplain ${f_or_d}.macports/*] {
+                    xinstall -m 0644 ${f} \
+                        ${f_or_d}/[file tail ${f}]
+                }
+            }
+        }
+    }
+
+    # calenderserver configuration and nginx reverse proxy
+    foreach f_or_d {
+        Config/calendarserver.plist
+        etc/nginx.conf
+    } {
+        install_initial_configuration \
+            ${calendarserverdir}/Library/CalendarServer/${f_or_d}
+    }
+
+    # configure all template files with local settings
+    foreach d_or_f {
+        Config
+        etc
+        } {
+        fs-traverse f ${calendarserverdir}/Library/CalendarServer/${d_or_f} {
+            if { [file isfile ${f}]
+                 && ![string match "*.macports" ${f}]
+                 && [string match "text/*" \
+                         [lindex [exec /usr/bin/file --mime-type ${f}] end]]
+             } then {
+                foreach cmd [list \
+                     "s|@PREFIX@|${prefix}|g" \
+                     "s|@host@.@domain@.@tld@|${fullhost}|g" \
+                     "s|@domain@.@tld@|${domaintld}|g" \
+                     "s|@host@|${host}|g" \
+                     "s|@domain@|${domain}|g" \
+                     "s|@tld@|${tld}|g" \
+                     "s|@TLS_CERTIFICATE_NAME@|${tls_certificate_name}|g" \
+                     "s|@POSTGRESQL9_VERSION_NICKNAME@|${postgresql9_version_nickname}|g" \
+                    ] {
+                    reinplace -q ${cmd} ${f}
+                }
+            }
+        }
+    }
+
+    # Add launchd.plist keys to MacPorts launchdaemons
+
+    # org.macports.calendarserver
+    system -W \
+        ${prefix}/etc/${startupitem.location}/org.macports.calendarserver \
+        "/usr/bin/plutil -insert EnvironmentVariables -xml \
+            \"<dict> \
+                <key>PYTHON</key> \
+                <string>${calendarserverdir}/Library/CalendarServer/${ccsname}/bin/python</string> \
+                <key>PATH</key> \
+                <string>${calendarserverdir}/Library/CalendarServer/${ccsname}/bin:${calendarserverdir}/Library/CalendarServer/${ccsname}/virtualenv/bin:${prefix}/bin:${prefix}/sbin:/usr/bin:/usr/sbin:/bin:/sbin</string> \
+            </dict>\" \
+            org.macports.calendarserver.plist"
+    system -W \
+        ${prefix}/etc/${startupitem.location}/org.macports.calendarserver \
+        "/usr/bin/plutil -insert UserName -string ${calendarserverUser} \
+            org.macports.calendarserver.plist"
```

Comment:
)What does this do? Replace variables in plist file? (If yes, couldn't macports use some reinplace?)

Reply:
These are all the key values used by Apple Server.app’s now-deprecated Calendar and Contacts Server.

`plutil` Is the native utility for editing plists. Using `reinplace` is hacky, but useful for changing single entries as done elsewhere. For entirely new entries required here, you want `plutil`.

Reply:
Yeah MacPorts doesn't have built-in support for doing this kind of plist editing, but the need has come up several times. Would be great if some day we added a MacPorts `plutil` Tcl command.

Reply:
See this ad hoc subroutine in PR https://github.com/macports/macports-ports/pull/5643:
```
proc plutil_startup {plcmds label} {
     global prefix startupitem.location
     foreach cmd ${plcmds} {
         system -W ${prefix}/etc/${startupitem.location}/${label} \
             "/usr/bin/plutil ${cmd} ${label}.plist"
     }
 }
```

I found this simple approach to be pretty useful and efficient for handling a lot of launch daemon edits.